tasks/hana_deploy.yml

```diff
@@ -1,18 +1,29 @@
 ---
 
-- name: Use SAPCAR to extract the SAP HANA Bundle SAR file
-  command: >-
-    {{ sap_hana_deployment_sapcar_path }}/{{ sap_hana_deployment_sapcar_file_name }} \
-    -xvf {{ sap_hana_deployment_bundle_path }}/{{ sap_hana_deployment_bundle_sar_file_name }} \
-    -manifest SIGNATURE.SMF
-  register: extractbundle
-  args:
-    chdir: "{{ sap_hana_deployment_sapcar_path }}"
-  changed_when: "'SAPCAR: processing archive' in extractbundle.stdout"
+- name: Unpack SAPCAR archive (backwards compatibility)
+  block:
+    - name: Use SAPCAR to extract the SAP HANA Bundle SAR file
+      command: >-
+        {{ sap_hana_deployment_sapcar_path }}/{{ sap_hana_deployment_sapcar_file_name }} \
+        -xvf {{ sap_hana_deployment_bundle_path }}/{{ sap_hana_deployment_bundle_sar_file_name }} \
+        -manifest SIGNATURE.SMF
+      register: sap_hana_deployment_register_extractbundle
+      args:
+        chdir: "{{ sap_hana_deployment_sapcar_path }}"
+      changed_when: "'SAPCAR: processing archive' in sap_hana_deployment_register_extractbundle.stdout"
+
+    - name: Setting fact for HANA installer path
+      set_fact:
+        sap_hana_installdir: "{{ sap_hana_deployment_sapcar_path }}/SAP_HANA_DATABASE"
+  when:
+    - not((sap_hana_deployment_bundle_path is undefined) or ( sap_hana_deployment_bundle_path is none) or (sap_hana_deployment_bundle_path | trim == '') )
+    - not((sap_hana_deployment_bundle_sar_file_name) or (sap_hana_deployment_bundle_sar_file_name) or (sap_hana_deployment_bundle_sar_file_name | trim == ''))
```

Comment:
You are checking this twice --> `sap_hana_deployment_bundle_sar_file_name`

Reply:
no. It'sthe only way to check if a variable is undefined, empty or emptied, i.e.contain whitespaces. I do this check for sap_hana_deployment_bundle_path is undefined and sap_hana_deployment_bundle_sar_file_name. 
Don't get your point here?

Reply:
please lets discuss

Reply:
@rhmk check the second conditional:
`- not((sap_hana_deployment_bundle_sar_file_name) or (sap_hana_deployment_bundle_sar_file_name) or (sap_hana_deployment_bundle_sar_file_name | trim == ''))`

There is no difference in how you are evaluating `sap_hana_deployment_bundle_sar_file_name` in the first 2 entries. For the third one there is a difference, but not for the others.

Reply:
Dump me. Got it. Fixed now :-)

Reply:
unfortunately the line is now 166 characters and I have no glue how to change that :-( 
ansible-lint complains
I could omit the first check "(sap_hana_deployment_bundle_sar_file_name) is undefined)" as it is defined in defaults. What do you suggest?

Reply:
for now I removed the first check,because the variable is always defined in deafults, but empty